### PR TITLE
feat:  Adds AMQP Inbound Transport

### PR DESCRIPTION
**Pull request by pfeairheller**
This PR included

- AMQP Inbound Transport (eg: RabbitMQ)

Signed-off-by: Phil Feairheller <pfeairheller@gmail.com>

### CODE REVIEW

**Comment by troyronda** [comment]
@pfeairheller do you have a suggestion on who could help review this PR?

File: .github/workflows/transport-amqp.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+          go-version: 1.15
+        id: go
+
+      - uses: actions/checkout@v2
+
+      - name: Run unit test
+        timeout-minutes: 15
+        run: make unit-test
+
+      - name: Upload coverage to Codecov
```
Comment (troyronda) [inline .github/workflows/transport-amqp.yml:52]: It seems that we didn't get a codecov comment for this PR.

File: .github/workflows/common.yml [hunk 2]
```diff
@@ -28,7 +28,14 @@ jobs:
   unitTest:
     name: Unit test
     runs-on: ubuntu-18.04
+    services:
+      rabbitmq:
+        image: rabbitmq
+        ports:
+          - 5672:5672
```
Comment (soluchok) [inline .github/workflows/common.yml:35]: why do we need this docker? we can use `dockertest`. See https://github.com/hyperledger/aries-framework-go-ext/blob/main/component/storage/mysql/store_test.go#L37

File: .github/workflows/transport-amqp.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
+#
+# Copyright SecureKey Technologies Inc. All Rights Reserved.
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+name: transport-amqp
+on:
+  push:
+    paths:
+      - 'component/didcomm/transport/amqp/**'
```
Comment (soluchok) [inline .github/workflows/transport-amqp.yml:10]: You need also to ignore this path for a common workflow.
See https://github.com/hyperledger/aries-framework-go-ext/blob/main/.github/workflows/common.yml#L10 and https://github.com/hyperledger/aries-framework-go-ext/blob/main/.github/workflows/common.yml#L15

File: .github/workflows/common.yml [hunk 4]
```diff
@@ -28,7 +28,14 @@ jobs:
   unitTest:
     name: Unit test
     runs-on: ubuntu-18.04
+    services:
+      rabbitmq:
```
Comment (troyronda) [inline .github/workflows/common.yml:32]: rabbitmq shouldn't be part of a common job.

File: .github/workflows/transport-amqp.yml (hunk 3 above)
Comment (pfeairheller) [reply .github/workflows/transport-amqp.yml:10]: Yeah, I noticed that pattern when doing the mongo driver.  I'll fix these.

File: .github/workflows/common.yml (hunk 4 above)
Comment (pfeairheller) [reply .github/workflows/common.yml:32]: Yep, this was my misunderstanding of the exclusion of component tests from common.  I'll get rid of it when I exclude the tests here.

File: .github/workflows/common.yml (hunk 2 above)
Comment (pfeairheller) [reply .github/workflows/common.yml:35]: Is `dockertest` a required approach?  The mongodb PR was approved and merged using this approach.

Comment (soluchok) [reply .github/workflows/common.yml:35]: I can't say that `dockertest ` is required. But it is better for sure. BTW `mongodb` tests don’t work. I will create an issue for that.

Comment (soluchok) [reply .github/workflows/common.yml:35]: https://github.com/hyperledger/aries-framework-go-ext/issues/22

Comment (soluchok) [reply .github/workflows/common.yml:35]: FYI: https://github.com/hyperledger/aries-framework-go-ext/issues/22

File: .github/workflows/transport-amqp.yml (hunk 1 above)
Comment (troyronda) [reply .github/workflows/transport-amqp.yml:52]: Perhaps a rebase will help.

File: .github/workflows/common.yml (hunk 2 above)
Comment (pfeairheller) [reply .github/workflows/common.yml:35]: I'll make the switch to both today, but I'm curious why `dockertest` is better.

Comment (soluchok) [reply .github/workflows/common.yml:35]: @pfeairheller `dockertest` simplifies the usage for users. You don't need to run or kill containers. Tests can be run easily.

Comment (pfeairheller) [reply .github/workflows/common.yml:35]: Ah, got it.  Makes it easier to run locally.  Makes sense, thanks for the answer.

Comment (pfeairheller) [reply .github/workflows/common.yml]: Done.

File: .github/workflows/transport-amqp.yml (hunk 1 above)
Comment (pfeairheller) [reply .github/workflows/transport-amqp.yml:47]: Looks like rebase did the trick, we have coverage

File: .github/workflows/common.yml [hunk 5]
```diff
@@ -31,6 +33,8 @@ jobs:
     name: Unit test
     runs-on: ubuntu-18.04
     timeout-minutes: 15
+    env:
+      RABBITMQ_HOST: localhost
```
Comment (troyronda) [inline .github/workflows/common.yml:37]: This is not a common environment variable.

File: .github/workflows/transport-amqp.yml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      - uses: actions/checkout@v2
+
+      - name: Run unit test
+        timeout-minutes: 15
+        run: make unit-test
+
+      - name: Upload coverage to Codecov
+        timeout-minutes: 10
+        if: github.repository == 'hyperledger/aries-framework-go-ext'
+        uses: codecov/codecov-action@v1.0.13
```
Comment (troyronda) [inline .github/workflows/transport-amqp.yml:50]: @soluchok does anything else need to be updated for codecov?

File: .github/workflows/common.yml (hunk 5 above)
Comment (pfeairheller) [reply .github/workflows/common.yml]: Fixed.

**Review by troyronda — APPROVED** [review]

**Pull request merged**